server/service/src/purchase_order/add_to_purchase_order_from_master_list.rs

```diff
@@ -0,0 +1,252 @@
+            .context(mock_store_a().id, "".to_string())
+            .unwrap();
+        let service = service_provider.purchase_order_service;
+
+        // PurchaseOrderDoesNotExist
+        assert_eq!(
+            service.add_to_purchase_order_from_master_list(
+                &context,
+                ServiceInput {
+                    purchase_order_id: "invalid".to_owned(),
+                    master_list_id: "n/a".to_owned()
+                },
+            ),
+            Err(ServiceError::PurchaseOrderDoesNotExist)
+        );
+
+        // CannotEditRecord
+        assert_eq!(
+            service.add_to_purchase_order_from_master_list(
+                &context,
+                ServiceInput {
+                    purchase_order_id: mock_purchase_order_c().id,
+                    master_list_id: "n/a".to_owned()
+                },
+            ),
+            Err(ServiceError::CannotEditPurchaseOrder)
+        );
+
+        // MasterListNotFoundForThisStore
+        assert_eq!(
+            service.add_to_purchase_order_from_master_list(
+                &context,
+                ServiceInput {
+                    purchase_order_id: mock_purchase_order_a().id,
+                    master_list_id: mock_test_not_store_a_master_list().master_list.id
+                },
+            ),
+            Err(ServiceError::MasterListNotFoundForThisStore)
+        );
+
+        // NotThisStore
+        context.store_id = mock_store_c().id;
+        assert_eq!(
+            service.add_to_purchase_order_from_master_list(
+                &context,
+                ServiceInput {
+                    purchase_order_id: mock_purchase_order_a().id,
+                    master_list_id: "n/a".to_owned()
+                },
+            ),
+            Err(ServiceError::NotThisStorePurchaseOrder)
+        );
+    }
+
+    #[actix_rt::test]
+    async fn add_from_master_list_success() {
+        let (_, _, connection_manager, _) = setup_all_with_data(
+            "purchase_order_add_from_master_list_success",
+            MockDataInserts::all(),
+            inline_init(|r: &mut MockData| {
+                r.full_master_lists = vec![mock_master_list_program()];
+            }),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager);
+        let context = service_provider
+            .context(mock_store_b().id, "".to_string())
+            .unwrap();
+        let service = service_provider.purchase_order_service;
+
+        let result: Vec<repository::PurchaseOrderLineRow> = service
+            .add_to_purchase_order_from_master_list(
+                &context,
+                ServiceInput {
+                    purchase_order_id: mock_purchase_order_d().id,
+                    master_list_id: mock_master_list_program().master_list.id,
+                },
+            )
+            .unwrap()
+            .into_iter()
+            .map(|line| line.purchase_order_line_row)
```

Comment:
only one line in this program, but leaving this infrastructure here in case we did want to test with a PO with multiple lines